src/crab/array_domain.cpp

```
@@ -332,78 +208,37 @@ std::vector<Cell> offset_map_t::get_overlap_cells(const offset_t o, const unsign
             if (x.overlap(o, size)) {
-                // FIXME: we might have some duplicates. this is a very drastic solution.
-                if (std::ranges::find(out, x) == out.end()) {
-                    out.push_back(x);
+                any_overlap = true;
+                if (x != *maybe_c) {
+                    if (std::ranges::find(out, x) == out.end()) {
+                        out.push_back(x);
+                    }
                 }
-                continue_outer_loop = true;
             }
```

Cppcheck (2.19.0)

[style] The function 'hash_value' is never used.

(unusedFunction)